### DISCUSSION TITLE
Ubuntu launching incorrect (old) OneDrive GUI at system startup

### DISCUSSION THREAD

**Discussion by cliverlong-1961** [bpozdena/OneDriveGUI#187] [Q&A] [answered]
When I boot Ubuntu 22.04, OneDriveGUI is automatically started and the cloud icon appears in the top status line.

When I click on show/hide on the cloud icon, the OneDriveGUI window that opens states the client is out-of-date and needs to be updated.

I can stop the OneDriveGUI from within the OpenDriveGUI window and restart OneDriveGUI with following command:

`~/OneDriveGUI/src$ python3 OneDriveGUI.py`

If I click on the new cloud in the status area after this command, the GUI states OneDrive sync is not running but there is no longer error message about an out-of-date client.

I then click on the triangle in the OneDriveGUI window and the message “Business shared folder has been deprecated “ is shown and the circle is red.

If I click on the triangle in the OneDriveGUI window, the triangle briefly tuns to a square, the circle goes from red to green, then in less than a second the square reverts to a triangle and the circle changes back from green to red.

So my first question is how do I get Ubuntu to run the latest version of OneDriveGUI client at startup?

Relevant messages from onedrive-gui.log follow:

> /tmp/onedrive-gui$ cat onedrive-gui.log 
> 2024-09-19 10:15:16,529 [OneDriveGUI.py:3093][fn=config_client_bin_path][INFO] - Onedrive client location: 'onedrive'
> 2024-09-19 10:15:16,598 [OneDriveGUI.py:2854][fn=get_installed_client_version][DEBUG] - [GUI] Installed client version is 250
> 
> 2024-09-19 10:15:17,550 [OneDriveGUI.py:3012][fn=main_window_start_state][INFO] - [GUI] Starting OneDriveGUI minimized to system tray
> 2024-09-19 10:15:17,551 [OneDriveGUI.py:1098][fn=__init__][INFO] - crlong_personal
> 2024-09-19 10:15:18,029 [connectionpool.py:1014][fn=_new_conn][DEBUG] - Starting new HTTPS connection (1): api.github.com:443
> 2024-09-19 10:15:18,142 [connectionpool.py:473][fn=_make_request][DEBUG] - https://api.github.com:443 "GET /repos/abraunegg/onedrive/releases/latest HTTP/1.1" 200 3285
> 2024-09-19 10:15:18,166 [OneDriveGUI.py:2472][fn=client_version_check][DEBUG] - [GUI] Client version check: {'latest_client_version': 'v2.5.0', 'installed_client_version': 'v2.5.0'}
> 2024-09-19 10:15:18,495 [OneDriveGUI.py:2545][fn=autostart_monitor][DEBUG] - [crlong_personal] Compatible client version found: False      **<<<<<<<<<<<<**
> 2024-09-19 10:15:18,496 [OneDriveGUI.py:2546][fn=autostart_monitor][DEBUG] - [crlong_personal] Auto-sync enabled for profile: False
> 2024-09-19 10:16:21,802 [OneDriveGUI.py:2428][fn=closeEvent][INFO] - [GUI] Minimizing main window to tray
> 2024-09-19 10:16:25,009 [OneDriveGUI.py:1150][fn=start_unsaved_changes_timer][DEBUG] - [GUI] Starting timer for unsaved changes checker
> 2024-09-19 10:16:29,810 [OneDriveGUI.py:1143][fn=stop_unsaved_changes_timer][DEBUG] - [GUI] Stopping timer for unsaved changes checker
> 2024-09-19 10:16:29,810 [OneDriveGUI.py:1118][fn=closeEvent][DEBUG] - [GUI] Closing Profiles window.
> 2024-09-19 11:05:10,462 [OneDriveGUI.py:2408][fn=graceful_shutdown][INFO] - Quitting OneDriveGUI

OneDrive GUI client stopped here then relaunched using

`~/OneDriveGUI/src$ python3 OneDriveGUI.py
`
OneDriveGUI log continues

> 2024-09-19 11:06:14,612 [OneDriveGUI.py:3150][fn=config_client_bin_path][INFO] - Onedrive client location: 'onedrive'
> 2024-09-19 11:06:14,621 [OneDriveGUI.py:2911][fn=get_installed_client_version][DEBUG] - [GUI] Installed client version is 250
> 2024-09-19 11:06:14,622 [OneDriveGUI.py:2946][fn=create_global_config][DEBUG] - [GUI] - loading default config {'

……

> 2024-09-19 11:06:14,623 [OneDriveGUI.py:2976][fn=create_global_config][DEBUG] - [GUI]{'crlong_personal': {'config_file': '/home/clive/.config/onedrive/accounts/crlong_personal/config', 'auto_sync': 'False', 
> …..
> 2024-09-19 11:06:15,245 [OneDriveGUI.py:3068][fn=main_window_start_state][INFO] - [GUI] Starting OneDriveGUI minimized to system tray
> 2024-09-19 11:06:15,247 [OneDriveGUI.py:1108][fn=__init__][INFO] - crlong_personal
> 2024-09-19 11:06:15,728 [connectionpool.py:1019][fn=_new_conn][DEBUG] - Starting new HTTPS connection (1): api.github.com:443
> 2024-09-19 11:06:15,862 [connectionpool.py:474][fn=_make_request][DEBUG] - https://api.github.com:443 "GET /repos/abraunegg/onedrive/releases/latest HTTP/1.1" 200 3285
> 2024-09-19 11:06:15,870 [OneDriveGUI.py:2444][fn=client_version_check][DEBUG] - [GUI] Client version check: Installed: v2.5.0 | Latest: v2.5.0
> 2024-09-19 11:06:16,194 [OneDriveGUI.py:2526][fn=autostart_monitor][DEBUG] - [crlong_personal] Compatible client version found: True   **<<<<<<<<<<<<**
> 2024-09-19 11:06:16,194 [OneDriveGUI.py:2529][fn=autostart_monitor][DEBUG] - [crlong_personal] Auto-sync enabled for profile: False
> 2024-09-19 11:06:31,124 [OneDriveGUI.py:1945][fn=__init__][INFO] - [GUI] Starting worker for profile crlong_personal
> 2024-09-19 11:06:31,124 [OneDriveGUI.py:1949][fn=__init__][DEBUG] - [GUI] OneDrive config file: /home/clive/.config/onedrive/accounts/crlong_personal/config
> ……
> 

So the lines I have highlighted with   **<<<<<<<<<<<< ** in the log above suggest initially on old version of OneDriveGUI is being loaded.

After I attempt to restart OneDrive from OneDriveGUI by clicking on the triangle I get the following log error messages:

> 2024-09-19 11:36:59,183 [OneDriveGUI.py:1945][fn=__init__][INFO] - [GUI] Starting worker for profile crlong_personal
> 2024-09-19 11:36:59,184 [OneDriveGUI.py:1949][fn=__init__][DEBUG] - [GUI] OneDrive config file: /home/clive/.config/onedrive/accounts/crlong_personal/config
> 2024-09-19 11:36:59,184 [OneDriveGUI.py:1950][fn=__init__][DEBUG] - [GUI] OneDrive config dir: <re.Match object; span=(0, 60), match='/home/clive/.config/onedrive/accounts/crlong_pers>
> 2024-09-19 11:36:59,184 [OneDriveGUI.py:1952][fn=__init__][DEBUG] - [GUI] Monitoring command: 'exec onedrive --confdir='/home/clive/.config/onedrive/accounts/crlong_personal' --monitor -v '
> 2024-09-19 11:36:59,184 [OneDriveGUI.py:2547][fn=start_onedrive_monitor][INFO] - [GUI] Opening external login window
> 2024-09-19 11:36:59,199 [OneDriveGUI.py:2022][fn=read_stdout][INFO] - [crlong_personal] D-Bus message bus daemon is available; GUI notifications are now enabled
> 2024-09-19 11:36:59,201 [OneDriveGUI.py:2022][fn=read_stdout][INFO] - [crlong_personal] Reading configuration file: /home/clive/.config/onedrive/accounts/crlong_personal/config
> 2024-09-19 11:36:59,201 [OneDriveGUI.py:2022][fn=read_stdout][INFO] - [crlong_personal] Invalid value for key in config file - using default value: monitor_interval
> 2024-09-19 11:36:59,201 [OneDriveGUI.py:2022][fn=read_stdout][INFO] - [crlong_personal] Invalid value for key in config file - using default value: monitor_fullscan_frequency
> 2024-09-19 11:36:59,201 [OneDriveGUI.py:2022][fn=read_stdout][INFO] - [crlong_personal] The option 'sync_business_shared_folders' has been deprecated and the process for synchronising Microsoft OneDrive Business Shared Folders has changed.
> 2024-09-19 11:36:59,201 [OneDriveGUI.py:2022][fn=read_stdout][INFO] - [crlong_personal] Please review the revised documentation on how to correctly configure this application feature.
> 2024-09-19 11:36:59,202 [OneDriveGUI.py:2022][fn=read_stdout][INFO] - [crlong_personal] You must update your client configuration and make changes to your local filesystem and online data to use this capability.
> 2024-09-19 11:36:59,202 [OneDriveGUI.py:2022][fn=read_stdout][INFO] - [crlong_personal] Configuration file has errors - please check your configuration
> 2024-09-19 11:37:00,804 [OneDriveGUI.py:2559][fn=<lambda>][INFO] - finished worker crlong_personal
> 2024-09-19 11:37:00,804 [OneDriveGUI.py:1956][fn=stop_worker][INFO] - [crlong_personal] Waiting for worker to finish...
> 2024-09-19 11:37:00,804 [OneDriveGUI.py:1960][fn=stop_worker][INFO] - [crlong_personal] Quitting thread
> 2024-09-19 11:37:00,805 [OneDriveGUI.py:1963][fn=stop_worker][INFO] - [crlong_personal] Removing thread info
> 2024-09-19 11:37:00,805 [OneDriveGUI.py:1966][fn=stop_worker][INFO] - [GUI] Remaining running workers: {}

OneDriveGUI config file contents:

> ~ /.config/onedrive/accounts/crlong_personal$ ls
> config  config_backup  items.sqlite3  items.sqlite3-shm  items.sqlite3-wal  refresh_token  sync_list
> clive@MacPro-Linux:~ /.config/onedrive/accounts/crlong_personal$ cat config
> sync_dir = "/home/clive/OneDrive"
> monitor_interval = "30"
> monitor_log_frequency = "5"
> monitor_fullscan_frequency = "10"
> skip_file = "~*|.~*|*.tmp"
> sync_business_shared_folders = "false"
> force_http_2 = "false"
> 

1. I can’t find information on how OneDriveGUI is being loaded at startup – if there are, can you direct me to them so I can ensure boot is always to the current version of OneDriveGUI?

2. Then find out why latest / correct OneDriveGUI is not able to run OneDrive.

Thanks.

**ADDED**

I don't know if the following is relevant 

I listed the OneDrive client config file and the initial lines are:

> clive@MacPro-Linux:/tmp/onedrive-gui$ onedrive --display-config
> D-Bus message bus daemon is available; GUI notifications are now enabled
> Application version                          = onedrive v2.5.0-1+np2+2.1
> Compiled with                                = LDC 2098
> User Application Config path                 = /home/clive/.config/onedrive
> System Application Config path               = /etc/onedrive
> Applicable Application 'config' location     = /home/clive/.config/onedrive/config
> Configuration file found in config location  = false - using application defaults
> Applicable 'sync_list' location              = /home/clive/.config/onedrive/sync_list
> Applicable 'items.sqlite3' location          = /home/clive/.config/onedrive/items.sqlite3

Notice the folder `/home/clive/.config/onedrive ` quoted in all the config paths above.

But when I list folder it contains

> ~ /.config/onedrive$ ls
> accounts  items.sqlite3  refresh_token

and `~ /.config/onedrive/accounts` contains `~ /.config/onedrive/accounts/crlong_personal`

```
~ /.config/onedrive/accounts/crlong_personal$ cat config
sync_dir = "/home/clive/OneDrive"
monitor_log_frequency = "5"
monitor_fullscan_frequency = "100"
skip_file = "~*|.~*|*.tmp"
sync_business_shared_folders = "false"
force_http_2 = "false"

```

There is no > Applicable Application 'config' location     = /home/clive/.config/onedrive/config

or any folder path

> /home/clive/.config/onedrive/config

the correct path is

> ~/.config/onedrive/accounts/

Is that relevant?

**Comment by bpozdena** [comment]
> When I boot Ubuntu 22.04, OneDriveGUI is automatically started and the cloud icon appears in the top status line.

OneDriveGUI does not have any buil-in functionality to autostart with your system. You either configured it manually or it was done by some 3rd party package/repo. You should remove it. 


> If I click on the new cloud in the status area after this command, the GUI states OneDrive sync is not running but there is no longer error message about an out-of-date client.

OneDriveGUI v1.1.0 is only compatible with OneDrive client v2.5.0. 


> I then click on the triangle in the OneDriveGUI window and the message “Business shared folder has been deprecated “ is shown and the circle is red.

Expected behavior. Read the documentation. 


> So my first question is how do I get Ubuntu to run the latest version of OneDriveGUI client at startup?

I'd suggest you to use the AppImage version of OneDriveGUI. I can't provide support for anything else. You can easily integrate AppImages using AppImageLauncher . 

Many desktop environments have some sort of GUI utility, which allows you to configure autostart applications. But if you integrated the AppImage using AppImageLauncer, it should have created a .desktop file in `~/.local/share/applications/` you might just need to copy it to  `~/.config/autostart`. Check your distro documentation on how to configure application autostart. 



You should also remove all your older versions of OneDriveGUI and OneDrive client that apparently are installed in your system.

**Reply by cliverlong-1961** [reply, marked as answer]
I will address the points in separate replies so this should make this easier to follow if someone has some or all of these issues in future.

> > When I boot Ubuntu 22.04, OneDriveGUI is automatically started and the cloud icon appears in the top status line.
> 
> OneDriveGUI does not have any buil-in functionality to autostart with your system. You either configured it manually or it was done by some 3rd party package/repo. You should remove it.
> 

Some web searching turned up the following which was the way to identify how the old version of oneDriveGUI was being started on the system and how to replace this with  the current 1.1 version.

Step 1: Open “Startup Applications” To open “Startup Applications” in Ubuntu, press Alt+F2 and run the `gnome-session-properties` command. Step 2: Add new startup application In the “Startup Applications Preferences” window that appears, click on the “Add” button to add a new startup application.

![image](https://github.com/user-attachments/assets/cc43d327-eced-405e-93b5-ba3e9233a711)

clive@MacPro-Linux:/usr/bin$ sudo cp ~/Downloads/OneDriveGUI-1.1.0-x86_64.AppImage /usr/bin/
[sudo] password for clive: 
clive@MacPro-Linux:/usr/bin$ ls | grep OneDriveGUI-1.
OneDriveGUI-1.0.3_fix116-x86_64.AppImage
OneDriveGUI-1.1.0-x86_64.AppImage

![image](https://github.com/user-attachments/assets/70a2fee0-22d7-4cd8-b007-1309d082f1d1)


> > If I click on the new cloud in the status area after this command, the GUI states OneDrive sync is not running but there is no longer error message about an out-of-date client.
> 
> OneDriveGUI v1.1.0 is only compatible with OneDrive client v2.5.0.
> 

I added this information about this message in the window to the original post to indicate what I saw as I was trying to fix the issue of the start-up problem. I added any information I believe was relevant to show my progress. As you have written this is expected behaviour - so, good.

So at this point at Ubuntu successfully starts up the latest version of the OneDriveGUI client: 1.1.0. Good

**Reply by cliverlong-1961** [reply]
Now my system is running the latest OneDriveGUI client, 1.1.0 which is communicating with the current OneDrive client, 2.5.0, I need to understand why the OneDriveGUI client does not interact with the OneDrive client.

Looking at the log file when OneDriveGUI was being started up there was a message that the sync_busines_shared_folders in the OneDrive client config file were deprecated and after a few more messages, the OneDriveGUI client failed to successfully "connect" to the OneDrive client.

The OneDrive client config file that had been created was as follows

> clive@MacPro-Linux:~ /.config/onedrive/accounts/crlong_personal$ cat config
> sync_dir = "/home/clive/OneDrive"
> enable_logging = "true"
> monitor_log_frequency = "5"
> monitor_fullscan_frequency = "100"
> skip_file = "~*|.~*|*.tmp"
> sync_business_shared_folders = "false"
> force_http_2 = "false"

Even though `sync_business_shared_folders` parameter was assigned the value "false" in the config file, I decided to comment it out resulting:

> clive@MacPro-Linux:~ /.config/onedrive/accounts/crlong_personal$ cat config
> sync_dir = "/home/clive/OneDrive"
> enable_logging = "true"
> monitor_log_frequency = "5"
> monitor_fullscan_frequency = "100"
> skip_file = "~*|.~*|*.tmp"
> '#' sync_business_shared_folders = "false"
> force_http_2 = "false"

I then stopped and started the OneDriveGUI client

> ~ /OneDriveGUI/src$ python3 OneDriveGUI.py 

And the OneDriveGUI client window no longer shows the message “Business shared folder has been deprecated “ 
See:

![image](https://github.com/user-attachments/assets/1d48c70b-de50-44f3-870e-2c04d64eae51)

The files are being checked by looking in the log `/tmp/onedrive-gui$ cat onedrive-gui.log `
See:

2024-09-19 15:40:10,768 [OneDriveGUI.py:2022][fn=read_stdout][INFO] - [crlong_personal] Including file - included by sync_list config:  ... file name ..

2024-09-19 15:40:10,768 [OneDriveGUI.py:2022][fn=read_stdout][INFO] - [crlong_personal] Including file - included by sync_list config: ... file name ..

2024-09-19 15:40:10,768 [OneDriveGUI.py:2022][fn=read_stdout][INFO] - [crlong_personal] ... file name ..

and so on for thousands of files.

**Reply by cliverlong-1961** [reply]
Can a version number be added to the OneDriveGUI window?

**Reply by bpozdena** [reply]
> Can a version number be added to the OneDriveGUI window?

Added to OneDriveGUI v1.1.1
